File: utils/py/algovivo/optimizer.py
class Optimizer:
    def __init__(
        self,
        max_iters=100,
        initial_step_size=1.0,
        backtracking_scale=0.3,
        max_line_search_iters=20,
        grad_q_tol=0.5 * 1e-5
    ):
        self.max_iters = max_iters
        self.initial_step_size = float(initial_step_size)
        self.backtracking_scale = float(backtracking_scale)
        self.max_line_search_iters = max_line_search_iters
        self.grad_q_tol = float(grad_q_tol)

    def to_step_args(self):
        return (
            self.max_iters,
            self.initial_step_size,
            self.backtracking_scale,
            self.max_line_search_iters,
            self.grad_q_tol
        )

File: algovivo/Optimizer.js
export default class Optimizer {
  constructor(args = {}) {
    this.maxIters = args.maxIters ?? 100;
    this.initialStepSize = args.initialStepSize ?? Math.fround(1);
    this.backtrackingScale = args.backtrackingScale ?? Math.fround(0.3);
    this.maxLineSearchIters = args.maxLineSearchIters ?? 20;
    this.gradQTol = args.gradQTol ?? Math.fround(0.5 * 1e-5);
  }

  toStepArgs() {
    return [
      this.maxIters,
      this.initialStepSize,
      this.backtrackingScale,
      this.maxLineSearchIters,
      this.gradQTol
    ];
  }

  dispose() {}
}
